modules/layers/src/contour-layer/contour-layer.js

```diff
@@ -192,15 +217,31 @@ export default class ContourLayer extends CompositeLayer {
     return strokeWidth;
   }
 
-  rebuildContours({oldProps, props}) {
-    if (oldProps.contours.length !== props.contours.length) {
+  shouldRebuildContours({oldProps, props}) {
+    if (
+      !oldProps.contours ||
+      !oldProps.zOffsetScale ||
+      oldProps.contours.length !== props.contours.length ||
```

Comment:
Is length comparison enough?

Reply:
In the code below, I am also checking for individual contour thresholds.

Also added triggers to detect cases when a color or strokeWidth is changed for a contour. Added unit tests